qualang_tools/plot/README.md

```diff
@@ -21,3 +21,72 @@ while job.result_handles.is_processing():
     ...
     # The execution will be cleanly interrupted by closing the figure
 ```
+
+## Fitting 
+This tool makes the fitting process much easier! Once you have the data you can:
+
+* Fit the data to a suitable fitting function.
+* Plot the data as well as the fitting function.
+* Extract physical parameters from the fitting.
+* Save the fitting parameters to a json file.
+* Open the json file and save its data to a python dictionary.
+
+### Use
+In order to use the fitting class, one should have `from fitting import *` in his/her code.
+ 
+### Usage example
+In the example below you can see five parts:
+1. The measured data - This section simulates the data in order to show the relevant functions.
+2. Fit - This section does the fit to the data.
+3. Plot - This section plots the data and the fitting plot.
+4. Save - This section saves the data to a json file. You are required to enter the file name, so the saved file will be of the form 'data_fit_{file_name}.json` 
+5. Read json - This section reads the json file and prints the saved parameters if the flag `print_params=True`.
+
+```python
+from fitting import *
+
+### Simulating the measured data ###
+# x and y are the measured data
+k = 10
+kc = 5
+A = 10
+
+x = np.linspace(-100, 100, 200)
+y = (
+        kc * (k / (1 + (4 / k ** 2) * (x + 10) ** 2))
+        + 1 * (np.random.rand(len(x)) - 0.5)
+        + 200
+)
+
+### Fit ###
+fit = Fit()
+# Choose the suitable fitting function
+fit_function = fit.transmission_resonator_spectroscopy(x, y)
+
+### Plot ###
+plt = Plot()
+# Change xlablel,ylabel and title as needed
+plt.plot(x, y, fit_function, xlabel="frequency[MHz]", ylabel="transmission[a.u.]", title="resonator spectroscopy power")
+
+#### Save ###
+save = Save()
+# Save the file under the name written in file_name. The saved file will be of the form 'data_fit_{file_name}.json`
+file_name = "transmission"
+save.save_params(x, y, fit_function, id=file_name)
+
+### Read json file as dictionary ###
+# In order to print the params use the flag `print_params=True`
+read = Read()
+data = read.read_saved_params(id=file_name, print_params=True)
+
+
+```
+
+Below is the plot of this example
+![fitting_example](example.PNG)
+
+# Possible future extensions

```

Comment:
Not sure if we want to list these, are we committing to adding them?

Reply:
I am about to work on those 3 statements... unless you think we don't need some of them..

Reply:
I would not write them in the readme.